.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+<!--
+Thank you for your pull request. Please provide a description above and review
+the requirements below.
+
+Bug fixes and new features should include tests and possibly benchmarks.
+
+Contributors guide: https://github.com/bcoe/c8/blob/master/CONTRIBUTING.md
+-->
+
+##### Checklist
+<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
+- [ ] `npm test`, tests passing
+- [ ] `npm run test:snap` (to update the snapshot)
+- [ ] tests and/or benchmarks are included
+- [ ] documentation is changed or added
+- [ ] commit message follows [conventional commit guidelines](https://www.conventionalcommits.org/)
```

Comment:
as mentioned below, this can be optional for the contributor and is really up to the merger to enforce with GH's squash and merge function.